instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/TripSessionsBillingTest.kt

```diff
@@ -0,0 +1,311 @@
+    }
+
+    @Test
+    fun testBillingSessionsForStoppedAndResumedActiveGuidanceTrip() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
+        mapboxNavigation.startTripSessionAndWaitForActiveGuidanceState()
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+
+        delay(TimeUnit.SECONDS.toMillis(SESSION_PAUSE_DURATION_SECONDS))
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
+        mapboxNavigation.startTripSessionAndWaitForActiveGuidanceState()
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(3, billingRequests.size)
+
+        assertTrue(billingRequests[0].isActiveGuidanceSession)
+        assertSessionDuration(0, billingRequests[0].duration)
+        assertTrue(billingRequests[1].isFreeDriveSession)
+        assertSessionDuration(SESSION_PAUSE_DURATION_SECONDS, billingRequests[1].duration)
+        assertTrue(billingRequests[2].isActiveGuidanceSession)
+        assertSessionDuration(0, billingRequests[2].duration)
+    }
+
+    private companion object {
+
+        const val HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS = 500L
+        const val SESSION_DURATION_SECONDS = 1L
+        const val SESSION_PAUSE_DURATION_SECONDS = 1L
+        const val ACTIVE_GUIDANCE_SKU_PREFIX = "10a"
+        const val FREE_DRIVE_SKU_PREFIX = "10b"
+
+        const val SESSIONS_ENDPOINT_PATH = "/sdk-sessions"
+
+        val HttpServiceEvent.isSessionEvent: Boolean
+            get() = url.path.startsWith(SESSIONS_ENDPOINT_PATH)
+
+        val HttpServiceEvent.skuParameter: String?
+            get() = url.parameters()["sku"]
+
+        val HttpServiceEvent.isActiveGuidanceSession: Boolean
+            get() = skuParameter?.startsWith(ACTIVE_GUIDANCE_SKU_PREFIX) == true
+
+        val HttpServiceEvent.isFreeDriveSession: Boolean
+            get() = skuParameter?.startsWith(FREE_DRIVE_SKU_PREFIX) == true
+
+        val HttpServiceEvent.duration: Long?
+            get() = url.parameters()["duration"]?.toLong()
+
+        fun HttpServiceEventsObserver.billingRequests(): List<HttpServiceEvent.Request> {
+            return onRequestEvents
+                .filter {
+                    it.isSessionEvent && (it.isFreeDriveSession || it.isActiveGuidanceSession)
+                }
+        }
+
+        private fun assertSessionDuration(expectedDurationSeconds: Long?, duration: Long?) {
+            if (expectedDurationSeconds == null) {
+                assertNull(duration)
+                return
+            }
+
+            assertNotNull("Session duration was null", duration)
+            requireNotNull(duration)
+
+            // Allow at most one second for session time calculation error
```

Comment:
I totally agree that we should use some tolerance here, but in all the cases above the possible expected values are only 0 and 1. So the tolerance wouldn't make a lot of sense.
Can we switch to at least 2 seconds delay to be more strict with our checks here?

Reply:
Not sure about 2 seconds, because it seems too lot for calculation errors. And it seems that we can use strict comparison here (`duration == expectedDurationSeconds`) because we compare integer numbers. @VysotskiVadim what do you think?

Reply:
During a call agreed to change session duration to 2 sends and leave 1 second tolerance for duration comparison.